Fix #115: JSON encoding in client API

cdamus opened the pull request:
- use *emfjson-jackson* mapper in the `DefaultJsonCodec`
- ensure IDREFs for internal cross-referencees by using a temporary resource when marshalling
- make safe copies of commands in the server when shipping them around to various components, so that the notification doesn’t end up with an empty list of `objectsToAdd`

cdamus committed:
Fix #115: JSON encoding in client API

- use emfjson-jackson mapper in the DefaultJsonCodec
- ensure IDREFs for interal cross-referencees by using a temporary resource
  when marshalling
- make safe copies of commands in the server when shipping them around
   to various components, so that the notification doesn’t end up with an
   empty list of objectsToAdd

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

edgarmueller commented on com.eclipsesource.modelserver.client/src/main/java/com/eclipsesource/modelserver/client/ModelServerClient.java:
```diff
@@ -283,6 +286,13 @@ private boolean isSupportedFormat(String format) {
+		// HREFs (e.g., "#//@objectsToAdd.0") which will not resolve in the Model Server
+		// on account of the resource with URI "" not existing
+        Resource cmdRes = new ResourceImpl(URI.createURI("$command.json"));
+        cmdRes.getContents().add(EcoreUtil.copy(command));
```
Could we move this bit into the encoder? Formatting seems off

edgarmueller commented on com.eclipsesource.modelserver.client/src/test/java/com/eclipsesource/modelserver/client/ModelServerClientTest.java:
```diff
@@ -229,15 +232,24 @@ public void configure() throws MalformedURLException, ExecutionException, Interr
+    	add.setOwner(flow);
+    	add.setFeature("nodes");
+    	add.getObjectsToAdd().add(CoffeeFactory.eINSTANCE.createAutomaticTask());
+    	add.getObjectValues().addAll(add.getObjectsToAdd());
```
Formatting

edgarmueller commented on com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/ModelControllerTest.java:
```diff
@@ -169,4 +174,39 @@ public void executeCommand() throws EncodingException, DecodingException {
+
+        // Unload the resource so that the "owner" is a proxy in
+        // the expected command as well as the actual
+        res.unload();
```
Is this still needed?

cdamus replied:
No, indeed.

cdamus committed:
Code review comments #115

- refactor temp resource into JsonCode codec to share the strategy
  (the XmiCodec already does it, but not on a copy, so fix that)
- formatting
- remove obsolete work-around of unloading resources in tests

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

cdamus committed:
Eliminate redundant copying before encoding

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>
